src/Renderer/Camera.js

```diff
@@ -100,4 +102,25 @@ Camera.prototype.box3SizeOnScreen = function box3SizeOnScreen(box3, matrixWorld)
     return tmp.box3;
 };
 
+ /**
+ * Test for collision between camera and geometry layer (DTM/DSM) to adjust camera position
+ * @param {view} view where we test the collision between geometry layers and the camera
+ * @param {minDistanceCollision} minDistanceCollision the minimum distance allowed between the camera and the surface
+ */
+Camera.prototype.adjustAltitudeToAvoidCollisionWithLayer = function adjustAltitudeToAvoidCollisionWithLayer(view, minDistanceCollision) {
+    const camLocation = new Coordinates(view.referenceCrs, view.camera.camera3D.position).as('EPSG:4326');
+    const elevationLayer = view.getLayers(layer => layer.type === 'geometry')[0];
```

Comment:
You should pass the `layer` as a parameter to this function.
It should be more robust than assuming that it's the first `geometry` one.

Reply:
So this also indicates that instead of storing a boolean `handleCollision` you probably want to store a list of layers that you want the camera to collide with.
So my earlier comment example code would become:
```
this.controls.collisionLayers = options.handleCollision ? [wgs84TileLayer] : [];
```

WDYT?